core/test/lib/navigation-error-test.js

```diff
@@ -66,18 +67,44 @@ describe('#getNetworkError', () => {
     expect(error.friendlyMessage).toBeDisplayString(/^Lighthouse was unable to reliably load/);
   });
 
-  it('fails when page returns with a 404', () => {
+  it('warns when page returns with a 404 with flag', () => {
     const url = 'http://the-page.com';
     const mainRecord = makeNetworkRequest();
     mainRecord.url = url;
     mainRecord.statusCode = 404;
-    const error = getAndExpectError(mainRecord);
+    const context = {
+      url,
+      networkRecords: [mainRecord],
+      warnings: [],
+      loadFailureMode: LoadFailureMode.warn,
+      ignoreStatusCode: true,
+    };
+
+    const error = getNetworkError(mainRecord, context);
+    expect(error).toBeUndefined();
+    expect(context.warnings[0]).toBeDisplayString(/^Lighthouse was unable to reliably load/);
+  });
+
+  it('fails when page returns with a 404 without flag', () => {
+    const url = 'http://the-page.com';
+    const mainRecord = makeNetworkRequest();
+    mainRecord.url = url;
+    mainRecord.statusCode = 404;
+    const context = {
+      url,
+      networkRecords: [mainRecord],
+      warnings: [],
+      loadFailureMode: LoadFailureMode.warn,
+    };
+
+    const error = getAndExpectError(mainRecord, context);
+    expect(error).toBeDefined();
     expect(error.message).toEqual('ERRORED_DOCUMENT_REQUEST');
     expect(error.code).toEqual('ERRORED_DOCUMENT_REQUEST');
-    expect(error.friendlyMessage).toBeDisplayString(/^Lighthouse was unable to reliably load.*404/);
+    expect(error.friendlyMessage).toBeDisplayString(/^Lighthouse was unable to reliably load/);
```

Comment:
The 404 is important IMO, otherwise this could pass for one of the non-status code related page load errors.

Reply:
It would also likely be helpful to add a `documentStatusCode` (or something similar) to track the result easily pragmatically.

Reply:
[`runWarnings` only accepts string type currently](https://github.com/GoogleChrome/lighthouse/blob/main/types/lhr/lhr.d.ts#L42). I think it would be a larger change to allow extra data fields on warnings. Seems out of scope for this PR.

FWIW the bad status code is also tracked by our `http-status-code` audit which is easier to track programatically.